client/src/pages/admin/MergeLocations.js

```diff
@@ -0,0 +1,516 @@
+      setLocation02(null) // workaround for map already initialized error
+      if (value) {
+        setTimeout(() => setLocation02(new Location(value)))
+      }
+    }
+  }
+
+  useBoilerplate({
+    pageProps: DEFAULT_PAGE_PROPS,
+    searchProps: DEFAULT_SEARCH_PROPS,
+    pageDispatchers
+  })
+
+  const map = useMemo(() => {
+    if (!mergedLocation) {
+      return ""
+    }
+    return (
+      <Leaflet
+        mapId="merged-location"
+        markers={[
+          {
+            id: "marker-merged-location",
+            name: _escape(mergedLocation.name) || "", // escape HTML in location name!
+            lat: Location.hasCoordinates(mergedLocation)
+              ? mergedLocation.lat
+              : null,
+            lng: Location.hasCoordinates(mergedLocation)
+              ? mergedLocation.lng
+              : null
+          }
+        ]}
+      />
+    )
+  }, [mergedLocation])
+
+  return (
+    <div>
+      <Messages error={saveError} />
+      <h2 className="form-header">Merge Locations Tool</h2>
+      <div style={{ display: "flex" }}>
+        <LocationColumn
+          location={location01}
+          setLocation={setLocationFn01}
+          setFieldValue={setFieldValue}
+          style={{ flex: "0 0 33%" }}
+        />
+        <div style={{ margin: "0 0.5%", flex: "0 0 33%" }}>
+          <h4
+            style={{
+              height: "39px",
+              margin: 0,
+              borderBottom: "1px solid #CCCCCC",
+              borderTop: "1px solid #CCCCCC",
+              display: "flex",
+              justifyContent: "space-between",
+              alignItems: "center"
+            }}
+          >
+            <small>
+              <Button
+                rightIcon="double-chevron-right"
+                intent="primary"
+                text="Use All"
+                onClick={() => setAllFields(location01)}
+                disabled={!location01 || !location02}
+              />
+            </small>
+            <span>Merged Location</span>
+            <small>
+              <Button
+                icon="double-chevron-left"
+                intent="primary"
+                text="Use All"
+                onClick={() => setAllFields(location02)}
+                disabled={!location01 || !location02}
+              />
+            </small>
+          </h4>
+          {(!location01 || !location02) && (
+            <div style={{ padding: "16px 5%" }}>
+              <Callout intent="warning">
+                Please select both locations to proceed...
+              </Callout>
+            </div>
+          )}
+          {location01 && location02 && !mergedLocation.name && (
+            <div style={{ padding: "16px 5%" }}>
+              <Callout intent="primary">
+                Please choose name to proceed...
+              </Callout>
+            </div>
+          )}
+          {location01 && location02 && mergedLocation.name && (
+            <>
+              <LocationField
```

Comment:
This is again very similar to a `LocationColumn` and `locations\Form`. Can we consolidate?

Reply:
This is the middle column in UI and it differs from `LocationColumn` in three ways which i think shouldn't be in `LocationColumn`. 
1. In `LocationColumn` an `AdvancedSelect` exists for location selection which we don't have in this column.
2. In `LocationColumn` we display "Use fields" buttons next to each field which is actually re-usable through `getActionButton` function. However, here we display different action buttons for each field which might add unnecessary complexity to `LocationColumn` if try to adapt it.
3. In this column we display a different header (with "Use All" buttons) and messages based on the state of the parent component (MergeLocations)

Consequently, trying to re-use `LocationColumn` here will result in high complexity with respect to minimal re-usability.